plug.vim

```diff
@@ -1394,9 +1398,13 @@ def esc(name):
 
 def nonblock_read(fname):
   """ Read a file with nonblock flag. Return the last line. """
-  fread = os.open(fname, os.O_RDONLY | os.O_NONBLOCK)
-  buf = os.read(fread, 100000).decode('utf-8', 'replace')
-  os.close(fread)
+  try:
+    nonblock_flag = not G_IS_WIN and os.O_NONBLOCK or 0
+    fread = os.open(fname, os.O_RDONLY | nonblock_flag)
+    buf = os.read(fread, 100000).decode('utf-8', 'replace')
+    os.close(fread)
+  except:
+    return ''
```

Comment:
Doesn't quite seem right to me, since now it violates my description of returning the last line of the file to be read. How about instead, we simply only make the call if we aren't on Windows shell.

- Undo the changes to nonblock_read function.
- Then change line 1339 to:
```python
          line = '' if G_IS_WIN else nonblock_read(tfile.name)
```